### PR TITLE
feat: implement TableContext with handling for rows + cells

### PR DESCRIPTION
**Pull request by ling1726**
Implements `TableContext` and handle the `size` and `noNativeElements`

Addresses #23983

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/bafbc65e62bb33c5f5f37a39a02d57ee75a7f4a6'>bafbc65e62bb33c5f5f37a39a02d57ee75a7f4a6</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-table/src/components/TableRow/useTableRowStyles.ts [hunk 1]
```diff
@@ -14,21 +14,34 @@ export const tableRowClassNames: SlotClassNames<TableRowSlots> = {
...
   },
+
+  small: {
+    minHeight: '34px',
+    ...shorthands.borderBottom('1px', 'solid', tokens.colorNeutralStroke2),
+  },
+
+  smaller: {
+    minHeight: '24px',
+    fontSize: '12px',
```
Comment (bsunderhus) [inline packages/react-components/react-table/src/components/TableRow/useTableRowStyles.ts:35]: No tokens for font-size?

**Review by bsunderhus — APPROVED** [review]

File: packages/react-components/react-table/src/components/TableRow/useTableRowStyles.ts (hunk 1 above)
Comment (ling1726) [reply packages/react-components/react-table/src/components/TableRow/useTableRowStyles.ts:35]: ah yeah, good point, I'll hunt them down

Comment (ling1726) [reply packages/react-components/react-table/src/components/TableRow/useTableRowStyles.ts]: I've replaced all spacing/font values with tokens. The row sizes don't exist as tokens

**Pull request merged**